stdlib/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,211 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/dependencies"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+func actionFromSeverity(severity string) string {
+	if strings.ToLower(severity) == "ok" {
+		return "resolve"
+	}
+	return "trigger"
+}
+
+var actionFromSeverityFluxFn = values.NewFunction(
+	"actionFromSeverity",
+	semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{"severity": semantic.String},
+		Required:   semantic.LabelSet{"severity"},
+		Return:     semantic.String,
+	}),
+	func(ctx context.Context, deps dependencies.Interface, args values.Object) (values.Value, error) {
+		v, ok := args.Get("severity")
+		if !ok {
+			return nil, fmt.Errorf("missing argument: severity")
+		}
+
+		if v.Type().Nature() == semantic.String {
+			return values.NewString(actionFromSeverity(v.Str())), nil
+		}
+		return nil, fmt.Errorf("could not parse severity")
+	},
+	false,
+)
```

Comment:
Based on this implementation I would expect this function can be defined in pure Flux, something like:

```
actionFromSeverity = (severity) => if strings.toLower(v:severity) == "ok"" then "resolve" else "trigger"
```

Can you change it to be pure Flux?